.github/workflows/build_and_deploy.yml

```diff
@@ -59,36 +59,57 @@ jobs:
     runs-on: ubuntu-latest
     needs: [data-validation]
     steps:
+    - id: files
+      uses: jitterbit/get-changed-files@v1
+    - run: |
+          for changed_file in steps.files.outputs.all; do
+            echo " ${changed_file} changed "
+          done
     - uses: actions/checkout@v1
     - name: Set up Python 3.7
       uses: actions/setup-python@v1
       with:
         python-version: 3.7
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade "pip<20"
-        pip install -r requirements-dev.txt
+          python -m pip install --upgrade "pip<20"
+          pip install -r requirements-dev.txt
     - name: Cross-validate NLU model
-      run: |
-        rasa test nlu -f 3 --cross-validation
-        python .github/workflows/format_results.py
+      if: |
+          contains(  steps.files.outputs.all, 'data/nlu/' ) 
+          || contains(  steps.files.outputs.all, 'config.yml' )
```

Comment:
still wondering if we should try to do NLU on a specific tag, bc anytime we add rasa x nlu annotations it will trigger this (actually anything from rasa x rn bc of the synonyms re-ordering bug)
though i think changes to the config make sense as always triggering it

Reply:
as long as anyone making changes will add the flag when required - since we're not in the habit of it that's the only reason I wonder if it's good to do that?

Reply:
I think (hope) that any changes big enough to NLU that would warrant the need for the flag would push the person to want to test those changes, and the point of this was to make it not happen all the time, so i think it's okay. I definitely think config should always cause it though (or at least the pipeline, but dont think we can get that specific about it)

Reply:
sure, it's been added =)